src/resources/views/crud/fields/select2_grouped.blade.php

```diff
@@ -1,14 +1,19 @@
 <!-- select2 -->
 @php
     $current_value = old_empty_or_null($field['name'], '') ??  $field['value'] ?? $field['default'] ?? '';
-    $field['allows_null'] = $field['allows_null'] ?? $crud->model::isColumnNullable($field['name']);
+    if(!empty($current_value)) {
+        if (is_a($current_value, \Illuminate\Support\Collection::class)) {
+           $current_value = ($current_value)->pluck((new $field['model'])->getKeyName(), $field['attribute']);
+        }
+    }
```

Comment:
I have a feeling this won't be needed after we settle https://github.com/Laravel-Backpack/CRUD/pull/4127#discussion_r797336307

Otherwise I'm ok with this.